src/tools/rados/rados.cc

```diff
@@ -2158,6 +2170,11 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
 	    if (l <= 17 ||
 		(0 != i->get_oid().compare(l-17, 17,".0000000000000000"))) continue;
 	  }
+          if (list_by_pgid) {
```

Comment:
also, instead of stating from the first oid in the pool, can we start from `pg_t::get_hobj_start()`, and `break` once the pgid does not match. i.e. pass `pgid.ps()` as the start hash to `nobjects_begin()`?